### PR TITLE
Clarify documentation for SSL on AWS.

### PR DESCRIPTION
**Pull request by adambkaplan**
Addresses comments in kubernetes/kubernetes#36845

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/3375)

### CODE REVIEW

File: docs/concepts/services-networking/service.md [hunk 1]
```diff
@@ -428,30 +428,26 @@ an ephemeral IP will be assigned to the loadBalancer. If the `loadBalancerIP` is
 cloud provider does not support the feature, the field will be ignored.
 
 #### SSL support on AWS
-For partial SSL support on clusters running on AWS, starting with 1.3 two
-annotations can be added to a `LoadBalancer` service:
+For partial SSL support on clusters running on AWS, the following annotations can be
```
Comment (bowei) [inline docs/concepts/services-networking/service.md]: Can you clarify what "partial SSL support" means?

File: docs/concepts/services-networking/service.md [hunk 2]
```diff
@@ -428,30 +428,26 @@ an ephemeral IP will be assigned to the loadBalancer. If the `loadBalancerIP` is
...
+For partial SSL support on clusters running on AWS, the following annotations can be 
+added to a `LoadBalancer` service:
 
-```
+```yaml
     metadata:
       name: my-service
       annotations:
         service.beta.kubernetes.io/aws-load-balancer-ssl-cert: arn:aws:acm:us-east-1:123456789012:certificate/12345678-1234-1234-1234-123456789012
+        service.beta.kubernetes.io/aws-load-balancer-backend-protocol: (https|http|ssl|tcp)
```
Comment (bowei) [inline docs/concepts/services-networking/service.md]: Make the yaml cut-and-pasteable without change

```
        # Can be one of {https,http,ssl,tcp}
        service.beta.kubernetes.io/aws-load-balancer-backend-protocol: https
```

File: docs/concepts/services-networking/service.md [hunk 3]
```diff
@@ -428,30 +428,26 @@ an ephemeral IP will be assigned to the loadBalancer. If the `loadBalancerIP` is
...
+It can be either a third party certificate uploaded to IAM, or one created within AWS Certificate Manager.
 
-```yaml
-    metadata:
-      name: my-service
-      annotations:
-         service.beta.kubernetes.io/aws-load-balancer-backend-protocol: (https|http|ssl|tcp)
-```
+`aws-load-balancer-backend-protocol` (added in version 1.3) specifies which protocol a pod speaks. For HTTPS and
+SSL, the ELB will expect the pod to authenticate itself over the encrypted connection.
```
Comment (bowei) [inline docs/concepts/services-networking/service.md]: What does "expect the pod to authenticate itself" mean? Do you mean "present a valid TLS certificate" or something else?

**Comment by MohamedBassem** [comment]
Has anyone tried this annotations combo with `ingress.kubernetes.io/force-ssl-redirect: "true"`? For some reason the `HTTP_X_FORWARDED_PROTO` seems to disappear with this setup.

File: docs/concepts/services-networking/service.md (hunk 3 above)
Comment (adambkaplan) [inline docs/concepts/services-networking/service.md]: Checked the AWS documentation - for HTTPS the pod will need its own certificate. You are then responsible for adding the public key of the certificate to the AWS ELB.

File: docs/concepts/services-networking/service.md (hunk 1 above)
Comment (adambkaplan) [inline docs/concepts/services-networking/service.md]: Unclear what the original author meant by "partial." Configuring full SSL to the backend pods is painful and may require work to be done outside of the Kubernetes CLI. Most use an ELB to terminate the SSL connection and forward via HTTP.

**Comment by chenopis** [comment]
@bowei PTAL

File: docs/concepts/services-networking/service.md (hunk 1 above)
Comment (bowei) [inline docs/concepts/services-networking/service.md]: Can we change the text to be more specific (e.g. the backend is using HTTP without TLS), remove `partial SSL support`

**Comment by beweinreich** [comment]
@MohamedBassem I updated my service to include the `service.beta.kubernetes.io/aws-load-balancer-backend-protocol: http` part and its now sending the X-Forwarded-Proto header.

**Comment by zacharysarah** [comment]
@adambkaplan @bowei :wave: Pinging for updates. Where are we with this PR?

**Comment by bowei** [comment]
LGTM, but someone with actual AWS experience should probably review as well.

**Comment by steveperry-53** [comment]
@adambkaplan @bowei @MohamedBassem Can you suggest someone who has AWS experience?

**Comment by xiangpengzhao** [comment]
@adambkaplan PR needs rebase.
@chrislovecnm might help another review.

**Pull request closed without merging**